SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -300,6 +301,14 @@ void USpatialReceiver::OnAddComponent(const Worker_AddComponentOp& Op)
 			}
 		}
 		return;
+	case SpatialConstants::PARTITION_COMPONENT_ID:
+		if (APlayerController* Controller = Cast<APlayerController>(PackageMap->GetObjectFromEntityId(Op.entity_id).Get()))
```

Comment:
Wouldn't this happen in a critical section before the player controller is created?

Reply:
I thought this needed to happen in `ReceiveActor`, where we would get the partition entity and pass it through to `PostSpawnPlayerController`.

Reply:
Ah yes, the partition component gets added after the controller entity is created (as the claim partition only happens after the entity is returned to the creating server), so in my test example it happened afterwards. You're right that a server checked out the playercontroller after it was fully setup will need it to be done in the CreateActor step.

Reply:
So for the server that spawns the player, wouldn't it already have registered the net connection in `CreateSpatialNetConnection`?

Reply:
It will yes, but this operation is idempotent so we don't need to special case it.

Reply:
And all servers that see the player controller before it's claimed by the client need to handle this here in the AddComponent op.